Source/DafnyCore/Compilers/Compiler-java.cs

```diff
@@ -2136,15 +2168,19 @@ private class GenericArrayElementLvalue : ILvalue {
             throw new cce.UnreachableException();
         }
       } else {
-        // TODO-RS: This doesn't handle strings printed out as part of datatypes
-        bool isString = arg.Type.AsSeqType != null &&
-                        arg.Type.AsSeqType.Arg.IsCharType;
         bool isGeneric = arg.Type.AsSeqType != null &&
                          arg.Type.AsSeqType.Arg.IsTypeParameter;
-        if (isString) {
+        if (arg.Type.IsStringType) {
           TrParenExpr(arg, wr, false, wStmts);
           wr.Write(".verbatimString()");
-        } else if (isGeneric) {
+        } else if (arg.Type.IsCharType) {
+          if (UnicodeChars) {
```

Comment:
Is the condition here that the Dafny char type be boxed? (Which happens in the unicode case)?

Reply:
Yup, it's boxing the raw `int` value as a `CodePoint`, which will print the right thing in its `toString()` method. But I realize now I should have put an explicit call to `toString()`, instead of assuming the calling context will.

But there's an even better and more efficient way: I can just call `Helpers.ToCharLiteral(value)` instead :)